java/source/cdata.rst

```diff
@@ -0,0 +1,507 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+================
+C Data Interface
+================
+
+Recipes related to how to exchange data using C Data Interface specification.
+
+.. contents::
+
+Python (Consumer) - Java (Producer)
+===================================
+
+    For Python Consumer and Java Producer, please consider:
+
+    - The Root Allocator should be shared for all memory allocations.
```

Comment:
should this be `RootAllocator`?

Reply:
Thanks, let me update that

Reply:
@davisusanibar if it is the case, there are a few instances below.